### PR TITLE
Move node-build to it's own component step

### PR DESCRIPTION
**Pull request by jamescgoertzen**

### CODE REVIEW

File: lib/clio_setup/component/node_build.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    def name
+      "node-build"
+    end
+
+    def version
+      "3.0.10"
+    end
+
+    def install
+      run("brew upgrade node-build")
```
[the remark this answers is not in the document]
Comment (jamescgoertzen) [reply lib/clio_setup/component/node_build.rb]: I was operating under the assumption that `node-build` is a dependency of `nodenv` but this is much safer.

File: lib/clio_setup/component/node_build.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      local_version = output("node-build --version").match(/^v(\S+)/).captures.first
+      compatible_version?(version, local_version)
+    end
+
+    def name
+      "node-build"
+    end
+
+    def version
+      "3.0.10"
```
Comment (npaufler) [inline lib/clio_setup/component/node_build.rb]: Is this right? brew upgrade node-build is giving me 3.0.20

Comment (jamescgoertzen) [reply lib/clio_setup/component/node_build.rb]: I have `3.0.10` but I'll upgrade and ensure that it works with `3.0.20`!

Comment (jamescgoertzen) [reply lib/clio_setup/component/node_build.rb]: Doesn't seem to cause any issues, so I'll change this to `3.0.20`

**Pull request merged**